src/main/java/teammates/common/util/ActivityLogEntry.java

```diff
@@ -160,7 +185,11 @@ public ActivityLogEntry(String servlet, String act, AccountAttributes acc, Strin
             email = acc.email;
         }
         
-        id = googleId + formatTimeForId(new Date(time));
+        if (googleId.contentEquals("Unknown") && unregisteredUserCourse != null && unregisteredUserEmail != null) {
```

Comment:
It is good to assign conditions like this to a local variable so that it can be given an explanatory name.